pkg/util/fedinformer/typedmanager/single-cluster-manager.go

```diff
@@ -206,3 +209,11 @@ func (s *singleClusterInformerManagerImpl) Context() context.Context {
 func (s *singleClusterInformerManagerImpl) GetClient() kubernetes.Interface {
 	return s.client
 }
+
+func (s *singleClusterInformerManagerImpl) SetTransformFunc(resource schema.GroupVersionResource, handler cache.TransformFunc) error {
+	resourceInformer, err := s.informerFactory.ForResource(resource)
+	if err != nil {
+		return err
+	}
+	return resourceInformer.Informer().SetTransform(handler)
```

Comment:
> Transform function must be set before starting the informer.

This function can't guarantee that.

Reply:
How about move TransformFunc as a ForResource/Lister parameter？